### ISSUE TITLE
`<source_location>`: Investigate EDG issues

### ISSUE THREAD

**Issue by StephanTLavavej** [microsoft/STL#2859]
We have a workaround for internal VSO-1285779 "REPORTED: EDG rejects `source_location` intrinsics within `consteval` functions" in the test:

https://github.com/microsoft/STL/blob/969b2e250df56690768fdf8344293624d6ab0d6c/tests/std/tests/P1208R6_source_location/test.cpp#L4

This was resolved as fixed on 2021-09-07 by internal VS-PR-348924 in VS 2022 17.0. However, attempting to remove this workaround with VS 2022 17.3 Preview 3 fails with numerous compiler errors. We need to investigate and report these issues.

**Comment by cpplearner** [comment]
EDG seems to have trouble computing the correct location.

```
D:\test>type test-edg-srcloc.cpp
#include <source_location>

template<unsigned> struct A {};

A<std::source_location::current().line()> x = 0;

D:\test>cl /std:c++20 /BE test-edg-srcloc.cpp
Microsoft (R) C/C++ Optimizing Compiler Version 19.33.31627.1 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

test-edg-srcloc.cpp
"test-edg-srcloc.cpp", line 5: error: no suitable constructor exists to convert
          from "int" to "A<26U>"
  A<std::source_location::current().line()> x = 0;
                                                ^
```

Note that it thinks `std::source_location::current().line()` at line 5 evaluates to `26U`, maybe because `current` is defined at line 26 in `<source_location>`.

**Comment by frederick-vs-ja** [comment]
It seems that this issue can be closed as the workaround was removed by #3866.

**Comment by StephanTLavavej** [comment]
Agreed. All remaining EDG checks in the test are either column number variation (which we don't care about) or are actually guarding an MSVC bug. Thanks!
